modules/dnn/src/dnn.cpp

```diff
@@ -4613,6 +4613,7 @@ string Net::Impl::dump()
             case DNN_TARGET_CUDA_FP16: out << "CUDA_FP16"; colorId = 6; break;
             // don't use default:
         }
+        CV_Assert(colorId < colors.size());
```

Comment:
BTW, this check can help in runtime only.
(no problem detection if there is no tests for this code)

Reply:
yes, i'd also rather favour something, that throws a compile time error instead.

unfortunately, all i could come up with is this:

    enum X {A,B,C,D,MAX_X};
    std::array<string,MAX_X> {"A","B","C","D","E"};

(works if there are excessive strings, but stays silent if there are not enough)

Reply:
If you want to add "MAX" value into enum then please use this approach to hide that from documentation:

https://github.com/opencv/opencv/blob/4.5.0/modules/calib3d/include/opencv2/calib3d.hpp#L474-L476